priam/src/main/java/com/netflix/priam/backupv2/LocalDBReaderWriter.java

```
@@ -0,0 +1,245 @@
+                .filter(localDBEntry -> (localDBEntry.getFileUploadResult().getLastModifiedTime().equals(fileUploadResult.getLastModifiedTime())))
+                .filter(localDBEntry -> (localDBEntry.getFileUploadResult().getCompression().equals(fileUploadResult.getCompression())))
+                .collect(Collectors.toList());
+
+        if (localDBEntries == null || localDBEntries.isEmpty())
+            return null;
+
+        if (localDBEntries.size() == 1) {
+            if (logger.isDebugEnabled())
+                logger.debug("Local entry found: {}", localDBEntries.get(0));
+
+            return localDBEntries.get(0);
+        }
+
+        throw new Exception("Unexpected behavior: More than one entry found in local database for the same file. FileUploadResult: " + fileUploadResult);
+    }
+
+    public Path getLocalDBPath(final FileUploadResult fileUploadResult) {
+        return Paths.get(configuration.getDataFileLocation(), LOCAL_DB, fileUploadResult.getKeyspaceName(), fileUploadResult.getColumnFamilyName(), PrefixGenerator.getSSTFileBase(fileUploadResult.getFileName().toFile().getName()) + ".localdb");
+    }
+
+    public void writeLocalDB(final Path localDBFile, final LocalDB localDB) throws Exception {
+        if (localDB == null || localDBFile == null || localDBFile.toFile().isDirectory())
+            throw new Exception("Invalid Arguments: localDbFile: " + localDBFile + ", localDB: " + localDB);
+
+        if (!localDBFile.getParent().toFile().exists())
+            localDBFile.getParent().toFile().mkdirs();
+
+        try (FileWriter writer = new FileWriter(localDBFile.toFile())) {
+            writer.write(localDB.toString());
+        }
+    }
+
+    public LocalDB readLocalDB(final Path localDBFile) throws Exception {
+        //Verify it is file and it exists.
+        if (localDBFile.toFile().isDirectory())
```

Do we need to throw this exception ourselves, or can we let Java throw an exception for us when we attempt to construct a FileReader below?